Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,423 @@
+#pragma once
+
+#include "Layer.h"
+#include "IpAddress.h"
+#include "MacAddress.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @class WireGuardLayer
+	 * Represents a WireGuard protocol layer
+	 */
+	class WireGuardLayer : public Layer
+	{
+	public:
+		/**
+		 * WireGuard message types
+		 */
+		enum class WireGuardMessageType
```

Comment:
I think we need to add an `Unknown` value in case we try to parse an invalid message